src/libarena/lib.rs

```diff
@@ -221,16 +227,18 @@ impl<T> TypedArena<T> {
                     self.end.set(last_chunk.end());
                     return;
                 } else {
+                    // If the previous chunk's capacity is less than HUGE_PAGE
+                    // bytes, then this chunk will be least double the previous
+                    // chunk's size.
                     new_capacity = last_chunk.storage.capacity();
-                    loop {
+                    if new_capacity < HUGE_PAGE / elem_size {
                         new_capacity = new_capacity.checked_mul(2).unwrap();
-                        if new_capacity >= currently_used_cap + n {
-                            break;
-                        }
                     }
+
+                    // Also ensure that this chunk can fit `n`.
+                    new_capacity = cmp::max(n, new_capacity);
```

Comment:
I think this should happen just before the `TypedArenaChunk::new` and the `cmp::max` below should be removed. This way we guarantee that it happens for both paths and documents this happening the other branch below, too.

Same thing for the other arena

Reply:
Good idea.